### ISSUE TITLE
[DOC]: cuProj Readme

### ISSUE THREAD

**Issue by jarmak-nv** [rapidsai/cuspatial#1206]
### How would you describe the priority of this documentation request

Critical (currently preventing usage)

### Is this for new documentation, or an update to existing docs?

New

### Describe the incorrect/future/missing documentation

cuProj is new 🎉 

We'll want a nice readme to help ensure users can take advantage of our new GPU accelerated transformations between WGS84 and UTM!

We shouldn't start work on this until we're certain of what will exactly make it into 23.08.

**Comment by harrism** [comment]
There's a placeholder readme in #1191 

It needs: 
* examples
* build instructions
* contributing instructions

The rest will come once there's a Python API

**Comment by jarmak-nv** [comment]
Yup! I just wanted to have an issue to make sure we don't lose track of the updates it needs, and to keep it in the effort estimations for 23.08
